include/core/Layer.hpp

```diff
@@ -0,0 +1,141 @@
+        */
+        time_t current_timestep_epoch_time() { return simulation_time.get_current_epoc_time(); }
+
+
+        /***
+         * @brief Return the numeric id of this layer
+        */
+        int get_id() const { return this->description.id; }
+
+        /***
+         * @brief Return the name of this layer
+        */
+        const std::string& get_name() const { return this->description.name; }
+
+        /***
+         * @brief Return this time_step interval used for this layer
+        */
+        const double get_time_step() const { return this->description.time_step; }
+
+        /***
+         * @brief Return the units for the time_step value of this layer
+        */
+        const std::string& get_time_step_units() const { return this->description.time_step_units; }
+
+        /***
+         * @breif Run one simulation timestep for each model in this layer
+        */
+        virtual void update_models()
+        {
+            auto idx = simulation_time.next_timestep_index();
+            auto step = simulation_time.get_output_interval_seconds();
+            
+            //std::cout<<"Output Time Index: "<<output_time_index<<std::endl;
+            if(output_time_index%100 == 0) std::cout<<"Running timestep " << output_time_index <<std::endl;
+            std::string current_timestamp = simulation_time.get_timestamp(output_time_index);
+            for(const auto& id : processing_units) 
+            {
+                int sub_time = output_time_index;
+                //std::cout<<"Running cat "<<id<<std::endl;
+                auto r = features.catchment_at(id);
+                //TODO redesign to avoid this cast
+                auto r_c = std::dynamic_pointer_cast<realization::Catchment_Formulation>(r);
+                double response = r_c->get_response(output_time_index, description.time_step);
+                std::string output = std::to_string(output_time_index)+","+current_timestamp+","+
+                                    r_c->get_output_line_for_timestep(output_time_index)+"\n";
+                r_c->write_output(output);
+                //TODO put this somewhere else.  For now, just trying to ensure we get m^3/s into nexus output
+                try{
+                    response *= (catchment_data->get_feature(id)->get_property("areasqkm").as_real_number() * 1000000);
+                }catch(std::invalid_argument &e)
+                {
+                    response *= (catchment_data->get_feature(id)->get_property("area_sqkm").as_real_number() * 1000000);
+                }
+                //TODO put this somewhere else as well, for now, an implicit assumption is that a modules get_response returns
+                //m/timestep
+                //since we are operating on a 1 hour (3600s) dt, we need to scale the output appropriately
+                //so no response is m^2/hr...m^2/hr * 1hr/3600s = m^3/hr
+                response /= 3600.0;
```

Comment:
So, there was no great place to have a timestep variable before... but now we have a Simulation_Time object with an `output_interval`... even if we assume that the output is m/h (not m/timestep!), shouldn't this response be divided by the number of timesteps per hour if the timestep is not = 1?

Reply:
this was fixed in my branch